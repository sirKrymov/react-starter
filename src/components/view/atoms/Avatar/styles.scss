@import '../../../../styles/mixins';
@import '../../../../styles/configs';

.avatar {
  border-radius: 50%;
  overflow: hidden;
  cursor: default;
  @include center;

  &--with-border {
    transition: border-color $anim-time-fast;
    border: 2px solid var(--borderPrimary);

    &:hover {
      border-color: var(--borderDark);
    }
  }

  &--with-click {
    cursor: pointer;
  }

  &--xs {
    min-height: 36px;
    min-width: 36px;
    height: 36px;
    width: 36px;
  }

  &--sm {
    min-height: 80px;
    min-width: 80px;
    height: 80px;
    width: 80px;
  }

  &--lg {
    min-height: 100px;
    min-width: 100px;
    width: 100px;
    height: 100px;
  }
}
