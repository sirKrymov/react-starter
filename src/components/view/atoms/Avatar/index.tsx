// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  withBorder?: boolean;
  className?: string;
  onClick?(): void;
  avatar?: string;
  size?: 'xs' | 'sm' | 'lg' | number;
  id: string;
}

export function Avatar({
  className,
  onClick,
  withBorder,
  avatar,
  size = 'xs',
  id
}: IProps): ReactElement {
  const avatarClass = cn(
    {
      avatar: true,
      'avatar--with-border': withBorder,
      'avatar--with-click': onClick,
      ...(typeof size === 'string' && { [`avatar--${size}`]: size })
    },
    className
  );

  return (
    <div
      className={avatarClass}
      onClick={onClick ? onClick : undefined}
      style={
        typeof size === 'number'
          ? {
              height: size,
              width: size
            }
          : {}
      }
      id={id}
    >
      <img
        className="img-cover"
        src={
          avatar ||
          'https://f0.pngfuel.com/png/980/886/male-portrait-avatar-png-clip-art.png'
        }
        alt="avatar"
      />
    </div>
  );
}
