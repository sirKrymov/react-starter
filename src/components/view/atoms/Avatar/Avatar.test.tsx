import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import { Avatar } from '../Avatar';

describe('<Avatar> atom component', () => {
  const handleClick = sinon.spy();
  let shallowComponent: ShallowWrapper;

  const createComponent = (props?: Record<string, any>) => {
    shallowComponent = shallow(<Avatar id="avatar" {...props} />);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });

  describe('WHEN the component have additional properties', () => {
    it('AND it should handle click logic', () => {
      createComponent({ onClick: handleClick });

      expect(shallowComponent.find('div.avatar--with-click')).toHaveLength(1);
      shallowComponent.find('div').simulate('click');
      expect(handleClick).toHaveProperty('callCount', 1);
    });

    it('AND it should handle size number logic', () => {
      createComponent({ size: 20 });

      expect(shallowComponent.find('#avatar').prop('style')).toHaveProperty(
        'width',
        20
      );
      expect(shallowComponent.find('#avatar').prop('style')).toHaveProperty(
        'height',
        20
      );
    });
  });
});
