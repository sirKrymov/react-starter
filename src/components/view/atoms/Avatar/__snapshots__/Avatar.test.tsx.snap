// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="avatar avatar--xs"
  id="avatar"
  style={Object {}}
>
  <img
    alt="avatar"
    className="img-cover"
    src="https://f0.pngfuel.com/png/980/886/male-portrait-avatar-png-clip-art.png"
  />
</div>
`;
